cubic-server/Player.cpp

```diff
@@ -1015,3 +880,175 @@ void Player::_processKeepAlive()
+            (uint8_t)protocol::PlayerAbilitiesFlags::CreativeMode,
+        0.05,
+        0.1
+    });
+
+    // TODO: set Held Item
+
+    // TODO: update recipes
+
+    // TODO: update tags
+
+    // TODO: entity event ?
+
+    // TODO: command list
+
+    // TODO: update recipes book
+
+    this->_synchronizePostion({8.5, 100, 8.5});
+
+    this->sendServerData({
+        false,
+        "",
+        false,
+        "",
+        false
+    });
+
+    _dim->addEntity(this);
+    LDEBUG("Added entity player to dimension");
+    getDimension()->getWorld()->sendPlayerInfoAddPlayer(this);
+
+    this->sendSetCenterChunk({0, 0});
+
+    auto renderDistance = this->getDimension()->getWorld()->getRenderDistance();
+    // Send all chunks around the player
+    // TODO: send chunk closer to the player first
+    sendChunkAndLightUpdate(0, 0);
+    for (int32_t x = -renderDistance; x < renderDistance + 1; x++) {
+        for (int32_t z = -renderDistance; z < renderDistance + 1; z++) {
+            if (x == 0 && z == 0)
+                continue;
+            sendChunkAndLightUpdate(x, z);
+        }
+    }
+    // for (auto &player : this->_player->getDimension()->getPlayerList())
+    //     player->_synchronizePostion({0, -58, 0});
+    // this->_player->sendChunkAndLightUpdate(0, 0);
+    getDimension()->spawnPlayer(this);
+    this->_synchronizePostion({8.5, 100, 8.5});
+
+    this->_sendLoginMessage();
+}
+
+void Player::_sendLoginMessage(void)
+{
+    // Send login message
+    chat::Message connectionMsg = chat::Message("", {
+        .color = "yellow",
+        .translate = "multiplayer.player.joined",
+        .with = std::vector<chat::Message>({
+            chat::Message(
+                this->getUsername(),
+                {
+                    .insertion = this->getUsername(),
+                },
+                chat::message::ClickEvent(
+                    chat::message::ClickEvent::Action::SuggestCommand,
+                    "/tell " + this->getUsername() + " "
+                ),
+                chat::message::HoverEvent(
+                    chat::message::HoverEvent::Action::ShowEntity,
+                    "{\"type\": \"minecraft:player\", \"id\": \"" + this->getUuidString() + "\", \"name\": \"" + this->getUsername() + "\"}"
+                )
+            )
+        })
+    });
+
+    this->getDimension()->getWorld()->getChat()->sendSystemMessage(
+        connectionMsg,
+        false,
+        this->getDimension()->getWorld()->getWorldGroup()
+    );
+}
+
+void Player::_unloadChunk(int32_t x, int32_t z) {
+    if (!this->_chunks.contains({x, z}))
+        return;
+    else if (this->_chunks[{x, z}] == ChunkState::Loading) {
+        this->_dim->removePlayerFromLoadingChunk({x, z}, this);
+        this->_chunks.erase({x, z});
+        return;
+    }
+
+    this->sendUnloadChunk(x, z);
+    this->_chunks.erase({x, z});
+}
+
+void Player::_synchronizePostion(Vector3<double> pos)
```

Comment:
`Postion` :)

Reply:
Also I would prefer if that function was called more like `teleport` or something like that.
Synchronize position is more just the protocol.

Reply:
but teleport is another paquet, this is why I said that I think we are using synchronizePosition badly
[teleport_entity](https://wiki.vg/index.php?title=Protocol&oldid=18067#Teleport_Entity)
[synchronize_position](https://wiki.vg/index.php?title=Protocol&oldid=18067#Synchronize_Player_Position)

Reply:
This is not a function to send a packet, that is a function to move the player somewhere else, therefore a method called teleport in the player would be appropriate in this case.